torchdynamo/symbolic_convert.py

```diff
@@ -442,15 +442,73 @@ def import_source(self, module_name):
         self.output.update_co_names(alias)
         return GlobalSource(alias)
 
+    def resolve_name(self, name, package, level):
+        """
+        Copied from the Cpython implementation of __import__
+        Resolve a relative module name to an absolute one.
+        """
+        bits = package.rsplit(".", level - 1)
+        if len(bits) < level:
+            raise ImportError("attempted relative import beyond top-level package")
+        base = bits[0]
+        return "{}.{}".format(base, name) if name else base
+
+    def calc_package(self):
+        """
+        Copied from the Cpython implementation of __import__
```

Comment:
I'd leave a link to the implementation in a link, makes it easy for people to crossref.